src/docker_hub.ml

```diff
@@ -22,18 +21,23 @@ let rec cmps = function
       | n -> n
 
 let hurl ~meth ~headers url =
-  Http_lwt_client.request
-    ~config:(`HTTP_1_1 Httpaf.Config.default) (* TODO: Remove this when https://github.com/roburio/http-lwt-client/issues/7 is fixed *)
-    ~meth
-    ~headers
-    url
-    (* TODO: This won't work once we handle things that aren't just short and simple JSON *)
-    (fun _ acc body -> Lwt.return (Some (Option.value ~default:"" acc ^ body)))
-    None
-  >|= function
-  | Ok ({Http_lwt_client.status = `OK; _}, Some body) -> Ok body
+  match
+    Httpcats.request
+      ~config:(`HTTP_1_1 H1.Config.default) (* TODO: Remove this when https://github.com/roburio/http-lwt-client/issues/7 is fixed *)
+      ~meth
+      ~headers
+      ~uri:url
+      (* TODO: This won't work once we handle things that aren't just short and simple JSON *)
+      ~f:(fun _ _ acc body -> match acc, body with
+        | None, None -> None
+        | Some x, None
+        | None, Some x -> Some x
+        | Some acc, Some body -> Some (acc ^ body))
+      None
+  with
+  | Ok ({Httpcats.status = `OK; _}, Some body) -> Ok body
   | Ok (resp, body) -> Error (`Api_error (resp, body))
-  | Error e -> Error e
+  | Error e -> Error (e : Httpcats.error :> [> fetch_errors])
```

Comment:
I feel like this should be improved in httpcats. `Httpcats.request` should return `(_, [> error]) result` instead of the non polymorphic version of that currently.

Reply:
You are absolutely right!